runtime/src/qio/auxFilesys/hdfs/qio_plugin_hdfs.c

```diff
@@ -147,10 +147,37 @@ qioerr hdfs_preadv (void* file, const struct iovec *vector, int count, off_t off
 
   STARTING_SLOW_SYSCALL;
 
+#ifdef HDFS3
+
+  DO_RETAIN(((hdfs_fs*)fs));
```

Comment:
Why is this DO_RETAIN here? I don't think you need to do anything to reference counts in this function.